### ISSUE TITLE
LIMIT query on DB2 modifies column positions in the resulting record set

### ISSUE THREAD

**Issue by morozov** [doctrine/dbal#2344]
Fetching a single column from a limited and non-limited queries produces different results. The table in the example contains a single record with a GUID-like ID.
```php
$query = 'SELECT id FROM users';
$stmt = $conn->query($query);
$all = $stmt->fetchAll(PDO::FETCH_COLUMN);
echo 'No LIMIT applied:', PHP_EOL;
var_dump($all);

$query = $conn->getDatabasePlatform()->modifyLimitQuery($query, 1, 0);
$stmt = $conn->query($query);
$all = $stmt->fetchAll(PDO::FETCH_COLUMN);
echo 'LIMIT applied:', PHP_EOL;
var_dump($all);
```
produces the following output:
```
No LIMIT applied:
array(1) {
  [0] =>
  string(36) "965e5ad6-fdbd-cf43-e75f-56faf36c7ee7"
}
LIMIT applied:
array(1) {
  [0] =>
  string(1) "1"
}
```

It happens because the `DC_ROWNUM` column in `DB2Platform::doModifyLimitQuery()` is prepended to the columns fetched from subquery, and thus shifts originally selected columns by one position right. If it was appended, the positions would remain the same.

**Comment by morozov** [comment]
Oops… seems to be fixed by e64c76d.

**Comment by Deltachaos** [comment]
@morozov  Can you close this issue then if it is fixed now?
